std/algorithm/comparison.d

```diff
@@ -1298,6 +1298,24 @@ MaxType!T max(T...)(T args)
     return cast(typeof(return)) (chooseB ? b : a);
 }
 
+auto max(alias keyGetter = (a) => a, Range)(Range r)
```

Comment:
return ElementType!Range instead of auto. That makes the documentation easier to read